rust/src/wrapper/doc_properties.rs

```diff
@@ -42,8 +48,12 @@ impl DocProperties {
     /// @returns {DocProperties} - The DocProperties object.
     #[wasm_bindgen(js_name = "setTitle", skip_jsdoc)]
     pub fn set_title(&self, title: &str) -> DocProperties {
+        let mut lock = self.inner.lock().unwrap();
+        let mut inner = std::mem::take(&mut *lock);
+        inner = inner.set_title(title);
+        let _ = std::mem::replace(&mut *lock, inner);
         DocProperties {
-            inner: self.clone().inner.set_title(title),
+            inner: Arc::clone(&self.inner),
         }
```

Comment:
Thank you! It looks good to me. Do you think we should consider adding macros to reduce this boilerplate?

Reply:
Thanks. modified to use macro.